README.rst
Python Client for Service Control API API
=========================================

|stable| |pypi| |versions|

`Service Control API API`_: 

- `Client Library Documentation`_
- `Product Documentation`_

.. |stable| image:: https://img.shields.io/badge/support-stable-gold.svg
   :target: https://github.com/googleapis/google-cloud-python/blob/main/README.rst#stability-levels
.. |pypi| image:: https://img.shields.io/pypi/v/google-cloud-service-control.svg
   :target: https://pypi.org/project/google-cloud-service-control/
.. |versions| image:: https://img.shields.io/pypi/pyversions/google-cloud-service-control.svg
   :target: https://pypi.org/project/google-cloud-service-control/
.. _Service Control API API: https://cloud.google.com/service-infrastructure/docs/overview/
.. _Client Library Documentation: https://cloud.google.com/python/docs/reference/servicecontrol/latest
.. _Product Documentation:  https://cloud.google.com/service-infrastructure/docs/overview/

Quick Start
-----------

In order to use this library, you first need to go through the following steps:

1. `Select or create a Cloud Platform project.`_
2. `Enable billing for your project.`_
3. `Enable the Service Control API API.`_
4. `Setup Authentication.`_

.. _Select or create a Cloud Platform project.: https://console.cloud.google.com/project
.. _Enable billing for your project.: https://cloud.google.com/billing/docs/how-to/modify-project#enable_billing_for_a_project
.. _Enable the Service Control API API.:  https://cloud.google.com/service-infrastructure/docs/overview/
.. _Setup Authentication.: https://googleapis.dev/python/google-api-core/latest/auth.html

Installation
~~~~~~~~~~~~

Install this library in a `virtualenv`_ using pip. `virtualenv`_ is a tool to
create isolated Python environments. The basic problem it addresses is one of
dependencies and versions, and indirectly permissions.

With `virtualenv`_, it's possible to install this library without needing system
install permissions, and without clashing with the installed system
dependencies.

.. _`virtualenv`: https://virtualenv.pypa.io/en/latest/


Code samples and snippets
~~~~~~~~~~~~~~~~~~~~~~~~~

Code samples and snippets live in the `samples/` folder.


Supported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^
Our client libraries are compatible with all current [active](https://devguide.python.org/devcycle/#in-development-main-branch) and [maintenance](https://devguide.python.org/devcycle/#maintenance-branches) versions of
Python.

Python >= 3.7

Unsupported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^^^
Python <= 3.6

If you are using an [end-of-life](https://devguide.python.org/devcycle/#end-of-life-branches)
version of Python, we recommend that you update as soon as possible to an actively supported version.


Mac/Linux
^^^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    source <your-env>/bin/activate
    <your-env>/bin/pip install google-cloud-service-control


Windows
^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    <your-env>\Scripts\activate
    <your-env>\Scripts\pip.exe install google-cloud-service-control

Next Steps
~~~~~~~~~~

-  Read the `Client Library Documentation`_ for Service Control API API
   to see other available methods on the client.
-  Read the `Service Control API API Product documentation`_ to learn
   more about the product and see How-to Guides.
-  View this `README`_ to see the full list of Cloud
   APIs that we cover.

.. _Service Control API API Product documentation:  https://cloud.google.com/service-infrastructure/docs/overview/
.. _README: https://github.com/googleapis/google-cloud-python/blob/main/README.rst


CHANGELOG.md
# Changelog

### [1.0.3](https://www.github.com/googleapis/python-service-control/compare/v1.0.2...v1.0.3) (2021-09-02)


### Documentation

* migrate links in CONTRIBUTING.rst from master to main ([#58](https://www.github.com/googleapis/python-service-control/issues/58)) ([98c4177](https://www.github.com/googleapis/python-service-control/commit/98c4177b4cdeb4cc6ad2ac947bff0bb22df578a6))

### [1.0.2](https://www.github.com/googleapis/python-service-control/compare/v1.0.1...v1.0.2) (2021-07-28)


### Bug Fixes

* enable self signed jwt for grpc ([#52](https://www.github.com/googleapis/python-service-control/issues/52)) ([2941b84](https://www.github.com/googleapis/python-service-control/commit/2941b84c3a50ada15605679adeb38d250b3310a3))


### Documentation

* add Samples section to CONTRIBUTING.rst ([#48](https://www.github.com/googleapis/python-service-control/issues/48)) ([1bbd444](https://www.github.com/googleapis/python-service-control/commit/1bbd4440166cfa72a647a0dccd15e91222b4051a))


### Miscellaneous Chores

* release as 1.0.2 ([#53](https://www.github.com/googleapis/python-service-control/issues/53)) ([0dd6658](https://www.github.com/googleapis/python-service-control/commit/0dd6658d4b526f0559acd2fc08af9f6549b62023))

### [1.0.1](https://www.github.com/googleapis/python-service-control/compare/v1.0.0...v1.0.1) (2021-07-21)


### Bug Fixes

* **deps:** pin 'google-{api,cloud}-core', 'google-auth' to allow 2.x versions ([#47](https://www.github.com/googleapis/python-service-control/issues/47)) ([5dd88d4](https://www.github.com/googleapis/python-service-control/commit/5dd88d43aae715f8a899a6dece2a5fe25863e7a9))

## [1.0.0](https://www.github.com/googleapis/python-service-control/compare/v0.3.0...v1.0.0) (2021-07-18)


### Miscellaneous Chores

* release as 1.0.0 ([#42](https://www.github.com/googleapis/python-service-control/issues/42)) ([7722bef](https://www.github.com/googleapis/python-service-control/commit/7722bef5c084f1e610469ec09023fb9c712e904c))

## [0.3.0](https://www.github.com/googleapis/python-service-control/compare/v0.2.0...v0.3.0) (2021-07-01)


### Features

* add always_use_jwt_access ([#36](https://www.github.com/googleapis/python-service-control/issues/36)) ([7c6c08f](https://www.github.com/googleapis/python-service-control/commit/7c6c08ff833f17cb3adbfd576b7c1107fde3c852))
* Added the gRPC service config for the Service Controller v1 API ([#38](https://www.github.com/googleapis/python-service-control/issues/38)) ([b95a767](https://www.github.com/googleapis/python-service-control/commit/b95a7673dd59b2b84cd8d203ea3c05fc08d8b59e))


### Bug Fixes

* disable always_use_jwt_access ([#41](https://www.github.com/googleapis/python-service-control/issues/41)) ([77350ec](https://www.github.com/googleapis/python-service-control/commit/77350ec06c0385384cbb1a8c99164a270bc58c95))


### Documentation

* omit mention of Python 2.7 in 'CONTRIBUTING.rst' ([#1127](https://www.github.com/googleapis/python-service-control/issues/1127)) ([#31](https://www.github.com/googleapis/python-service-control/issues/31)) ([57502d9](https://www.github.com/googleapis/python-service-control/commit/57502d9d8b0cf68df2ab0f99b8bb4dcaeef7ae1f)), closes [#1126](https://www.github.com/googleapis/python-service-control/issues/1126)

## [0.2.0](https://www.github.com/googleapis/python-service-control/compare/v0.1.0...v0.2.0) (2021-05-27)


### Features

* bump release level to production/stable ([#25](https://www.github.com/googleapis/python-service-control/issues/25)) ([ce65557](https://www.github.com/googleapis/python-service-control/commit/ce655570ca4a8bafc93d55401b3e4ae47a747afb))
* support self-signed JWT flow for service accounts ([0630fa0](https://www.github.com/googleapis/python-service-control/commit/0630fa0538a633297cae935bd094443d7eeb8d45))


### Bug Fixes

* add async client to %name_%version/init.py ([0630fa0](https://www.github.com/googleapis/python-service-control/commit/0630fa0538a633297cae935bd094443d7eeb8d45))

## 0.1.0 (2021-04-03)


### Features

* generate v1 ([24f2e4a](https://www.github.com/googleapis/python-service-control/commit/24f2e4a943f2c80dc75fd34550a64dc5e025aa1c))


owl-bot-staging/v1/noxfile.py
# -*- coding: utf-8 -*-
# Copyright 2020 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#
import os
import pathlib
import shutil
import subprocess
import sys


import nox  # type: ignore

CURRENT_DIRECTORY = pathlib.Path(__file__).parent.absolute()

LOWER_BOUND_CONSTRAINTS_FILE = CURRENT_DIRECTORY / "constraints.txt"
PACKAGE_NAME = subprocess.check_output([sys.executable, "setup.py", "--name"], encoding="utf-8")


nox.sessions = [
    "unit",
    "cover",
    "mypy",
    "check_lower_bounds"
    # exclude update_lower_bounds from default
    "docs",
]

@nox.session(python=['3.6', '3.7', '3.8', '3.9'])
def unit(session):
    """Run the unit test suite."""

    session.install('coverage', 'pytest', 'pytest-cov', 'asyncmock', 'pytest-asyncio')
    session.install('-e', '.')

    session.run(
        'py.test',
        '--quiet',
        '--cov=google/cloud/servicecontrol_v1/',
        '--cov-config=.coveragerc',
        '--cov-report=term',
        '--cov-report=html',
        os.path.join('tests', 'unit', ''.join(session.posargs))
    )


@nox.session(python='3.7')
def cover(session):
    """Run the final coverage report.
    This outputs the coverage report aggregating coverage from the unit
    test runs (not system test runs), and then erases coverage data.
    """
    session.install("coverage", "pytest-cov")
    session.run("coverage", "report", "--show-missing", "--fail-under=100")

    session.run("coverage", "erase")


@nox.session(python=['3.6', '3.7'])
def mypy(session):
    """Run the type checker."""
    session.install('mypy', 'types-pkg_resources')
    session.install('.')
    session.run(
        'mypy',
        '--explicit-package-bases',
        'google',
    )


@nox.session
def update_lower_bounds(session):
    """Update lower bounds in constraints.txt to match setup.py"""
    session.install('google-cloud-testutils')
    session.install('.')

    session.run(
        'lower-bound-checker',
        'update',
        '--package-name',
        PACKAGE_NAME,
        '--constraints-file',
        str(LOWER_BOUND_CONSTRAINTS_FILE),
    )


@nox.session
def check_lower_bounds(session):
    """Check lower bounds in setup.py are reflected in constraints file"""
    session.install('google-cloud-testutils')
    session.install('.')

    session.run(
        'lower-bound-checker',
        'check',
        '--package-name',
        PACKAGE_NAME,
        '--constraints-file',
        str(LOWER_BOUND_CONSTRAINTS_FILE),
    )

@nox.session(python='3.6')
def docs(session):
    """Build the docs for this library."""

    session.install("-e", ".")
    session.install("sphinx<3.0.0", "alabaster", "recommonmark")

    shutil.rmtree(os.path.join("docs", "_build"), ignore_errors=True)
    session.run(
        "sphinx-build",
        "-W",  # warnings as errors
        "-T",  # show full traceback on exception
        "-N",  # no colors
        "-b",
        "html",
        "-d",
        os.path.join("docs", "_build", "doctrees", ""),
        os.path.join("docs", ""),
        os.path.join("docs", "_build", "html", ""),
    )
